### DISCUSSION TITLE
Ability to enter classes or id in the Interface and Display fields.

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#19364] [Feature Requests]
### Summary

If we could add classes to the elements that would greatly help for customization.

### Basic Example

<div data-v-d0788894="" data-v-7cec1471="" class="field half myClass1">
   <div data-v-0ac5ba37="" data-v-d0788894="" class="v-menu">...</div>
   <div data-v-03528dd7="" data-v-d0788894="" class="interface myClass2">...</div>
   <div data-v-7d9eb308="" data-v-ac52f53d="" data-v-d0788894="" class="v-dialog myClass3"></div>
</div>

### Motivation

There is no easy way to identify the divs

For instance, I want to identify the following div

```
<div data-v-d0788894="" data-v-7cec1471="" class="field half">
   <div data-v-0ac5ba37="" data-v-d0788894="" class="v-menu">...</div>
   <div data-v-03528dd7="" data-v-d0788894="" class="interface">...</div>
   <div data-v-7d9eb308="" data-v-ac52f53d="" data-v-d0788894="" class="v-dialog"></div>
</div>

```
but 
data-v-d0788894: is present  34 times
data-v-7cec1471: is present  8 times
data-v-03528dd7: is present  13 times
data-v-7d9eb308: is present  16 times
data-v-ac52f53d: is present  7 times

in a short form of about 10 elements

### Detailed Design

Add a field for Id and/or a field for classes
in the Interface and the Display Field

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

--

### Alternatives

--

### Adoption Strategy

--

**Comment by DanielBiegler** [comment]
Hi there! :)

Could you please provide an example customization you'd like to achieve? The docs on [theming](https://docs.directus.io/extensions/themes.html#custom-css) might help too.

Let's say I want to target the second field like so:

![image](https://github.com/directus/directus/assets/14810858/d6bce3aa-e398-4259-8876-95a355027163)

I can use the following CSS selector: `.v-form > .field.full:nth-child(2)`

Then you could make it more specific relating to other fields but I'm not quite sure what you're after so I'll leave it at that.

**Reply by Pachat** [reply]
Thank you for your reply. I'm also a fan of custom CSS.
But with many forms in my application, I tend to put only general rules or on the other side self-defined rules avoiding intermediate rules based on nth-child.
Because the rule will break if I change the order of fields or insert a new field before.
Maintaining such rules is not an option with dozens of forms.

That's why I'm looking for self-defined options like id or classes.
Also, it would help the stability of e2e automatic tests enormously.

**Comment by Pachat** [comment]
As we have collection and field for inputs which is enough to get the right element, the only missing part is for relational fields.
Therefore, I opened a new discussion #19372 and close this one.
